### PR TITLE
Add support for status property of azurerm_eventhub

### PR DESCRIPTION
**Pull request by neil-yechenwei**
Currently azurerm_eventhub doesn't support to set event hub status. So I submitted this PR to implement it.

--- PASS: TestAccEventHub_standard (306.97s)
--- PASS: TestAccEventHub_basic (370.29s)
--- PASS: TestAccEventHub_requiresImport (396.79s)
--- PASS: TestAccEventHub_basicOnePartition (415.47s)
--- PASS: TestAccEventHub_eventhubStatus (426.89s)
--- PASS: TestAccEventHub_messageRetentionUpdate (433.99s)
--- PASS: TestAccEventHub_captureDescription (437.25s)
--- PASS: TestAccEventHub_partitionCountUpdate (500.02s)
--- PASS: TestAccEventHub_captureDescriptionDisabled (529.67s)

API spec:
https://github.com/Azure/azure-rest-api-specs/blob/916568aafe943c0be88e7f845d38de5f189d4b68/specification/eventhub/resource-manager/Microsoft.EventHub/preview/2021-01-01-preview/eventhubs.json#L310

### CODE REVIEW

File: azurerm/internal/services/eventhub/eventhub_resource.go [hunk 1]
```diff
@@ -229,6 +251,10 @@ func resourceEventHubRead(d *pluginsdk.ResourceData, meta interface{}) error {
 			d.Set("message_retention", props.MessageRetentionInDays)
 			d.Set("partition_ids", props.PartitionIds)
 
+			if eventhubStatus := props.Status; eventhubStatus != nil {
+				d.Set("status", eventhubStatus)
```
Comment (mbfrahry) [inline azurerm/internal/services/eventhub/eventhub_resource.go:255]: I think we should take care of this lint error and then we'll be good to go

**Review by mbfrahry — CHANGES_REQUESTED** [review]
This is mostly good with one minor comment!

File: azurerm/internal/services/eventhub/eventhub_resource.go [hunk 2]
```diff
@@ -144,6 +144,23 @@ func resourceEventHub() *pluginsdk.Resource {
...
+					string(eventhubs.EntityStatusCreating),
+					string(eventhubs.EntityStatusDeleting),
+					string(eventhubs.EntityStatusDisabled),
+					string(eventhubs.EntityStatusReceiveDisabled),
+					string(eventhubs.EntityStatusRenaming),
+					string(eventhubs.EntityStatusRestoring),
+					string(eventhubs.EntityStatusSendDisabled),
+					string(eventhubs.EntityStatusUnknown),
+				}, false),
+			},
```
Comment (katbyte) [inline azurerm/internal/services/eventhub/eventhub_resource.go:162]: what happened when you set this property to creating or deleting? this doesn't seem likea useful property that makes sense to set? more something the service exposes?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: azurerm/internal/services/eventhub/eventhub_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply azurerm/internal/services/eventhub/eventhub_resource.go]: After checked, Azure portal and [azure cli](https://docs.microsoft.com/zh-cn/cli/azure/eventhubs/eventhub?view=azure-cli-latest) only exposed the `Active`, `Disabled` and `SendDisabled`. I assume they are more useful to user. So I've updated code to only support them.
image.

File: azurerm/internal/services/eventhub/eventhub_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply azurerm/internal/services/eventhub/eventhub_resource.go:249]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte @mbfrahry , thanks for your comments. I've updated code. Please have an another look. Thanks.

File: azurerm/internal/services/eventhub/eventhub_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply azurerm/internal/services/eventhub/eventhub_resource.go]: @katbyte , after confirmed with service team, `Active`, `Disabled` and `SendDisabled` are the only states that the service currently uses for Microsoft.EventHub/namespaces/eventHubs resource. So we just need to support these three values.

**Review by katbyte — APPROVED** [review]
Thanks @neil-yechenwei - LGTM 🚀

**Pull request merged**